### ISSUE TITLE
None of the `--log-level` cli options work

### ISSUE THREAD

**Issue by sbquinlan** [pangeo-forge/pangeo-forge-runner#121]
## What I did:

I wanted to change the default log level. I saw some properties on `BaseCommand` in the source so I looked at the help to see how to set them:

```bash
% pangeo-forge-runner --help                                                                             
# ... blah blah blah ...
--log-level=<Enum>
    Set the log level by value or name.
    Choices: any of [0, 10, 20, 30, 40, 50, 'DEBUG', 'INFO', 'WARN', 'ERROR', 'CRITICAL']
    Default: 30
    Equivalent to: [--Application.log_level]
```

Cool I'll set that:

```bash
% pangeo-forge-runner bake --log-level=CRITICAL
```

## What happened:

```bash
[Bake] WARNING | Config option `log_level` not recognized by `Bake`.
[Bake] WARNING | Config option `log_level` not recognized by `Bake`.
# ... followed by info logs
```

I tried `--log-level=0`, `--Application.log_level=WARN` and got more or less the same result

## What I expected

Bake accepts the log level config and does not INFO log.

**Comment by ranchodeluxe** [comment]
welp, this didn't do it either `--Bake.logging_config='{"version": 1, "loggers": {"pangeo_forge_recipes":{"level": "CRITICAL", "handlers":[], "propagate": False}}}'`

**Comment by ranchodeluxe** [comment]
Even this didn't work 😠 

```python
@dataclass
class ShutUpAndStoreToZarr(StoreToZarr):
    def expand(self, pcoll: beam.PCollection) -> beam.PCollection:
        rechunking_logger = logging.getLogger('pangeo_forge_recipes.rechunking')
        rechunking_logger.setLevel(logging.CRITICAL)
        return pcoll | StoreToZarr(store_name=self.store_name, combine_dims=self.combine_dims, target_chunks=self.target_chunks)

```

**Comment by yuvipanda** [comment]
The default logging config is for *traitlets*, and anything using `self.log` from objects that inherit from `LoggingConfigurable`.

I remember fiddling with it in https://github.com/pangeo-forge/pangeo-forge-runner/pull/65, which may be of use :)

**Comment by ranchodeluxe** [comment]
> The default logging config is for _traitlets_, and anything using `self.log` from objects that inherit from `LoggingConfigurable`.
> 
> I remember fiddling with it in #65, which may be of use :)

Right, what we're really after in this issue (which isn't explicity) is how to turn off logs in the recipes b/c @sbquinlan was thinking they are causing issues with some grpc calls (at least I think that's what he was saying in DMs)

**Comment by sbquinlan** [comment]
> what we're really after in this issue (which isn't explicit)

Actually I think the explicit problem that I described at the start is good for this issue. The _traitlet_ option for setting the log level does not seem to work at all. Is `--log-level` supposed to work?

> This also changes the default local one from multi_processing to multi_threading. The primary use here is to make the logging config from the runner be inherited by the workers too, which helps with debugging quite a bit.

Is this saying that, when running locally, Bake.log_level should propagate to the workers? That's cool, but I'm not sure I saw that happening. It's unclear from your PR how that was tested for me to have a workaround here.
